odoo/client.go

```diff
@@ -0,0 +1,108 @@
+package odoo
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+)
+
+// Client is the base struct that holds information required to talk to Odoo
+type Client struct {
+	baseURL string
+	db      string
+	http    *http.Client
+}
+
+// NewClient returns a new client with its basic fields set
+func NewClient(baseURL, db string) *Client {
```

Comment:
Smaller nit: In my experience with Go, at some point someone needs to set the http client by themselves. I'd love to see an optional argument where I can pass my own client.

Reply:
I guess so. Question is: Do we need to do it now? I'd rather avoid programming for unused eventuality as long there's no reason right now. It's not a library :)

Reply:
As long as `DefaultTransport` is used (it is) i'm happy for now 😉